### ISSUE TITLE
Attempt to return a null void pointer causes run-time error

### ISSUE THREAD

**Issue by lyskov** [pybind/pybind11#149]
Hi, i tested the recent commit that allows bindings of void* types and noticed that returning a nullptr causes a runtime error (while returning non-null pointer works fine). Please see example code that trigger this below.

Thanks,

```c++
#include <pybind11/pybind11.h>
#include <iostream>

void * foo_null_0() { return nullptr; }
void * foo_null_1() { return (void*)1234; }
void foo_null(void *p) { std::cout << p << std::endl; }

namespace py = pybind11;

PYBIND11_PLUGIN(example) {
    py::module m("example", "pybind11 example plugin");

	m.def("foo_null_0", &foo_null_0, "doc"); // invoking this in python yied: RuntimeError: Could not allocate capsule object!
	m.def("foo_null_1", &foo_null_1, "doc"); // invoking this works fine!
	m.def("foo_null", (void (*)(void*)) &foo_null, "doc", pybind11::arg(""));

    return m.ptr();
}
```

**Comment by wjakob** [comment]
fixed in 0772967.

**Comment by lyskov** [comment]
Thank you!
